src/main/java/com/powsybl/openloadflow/network/util/GenerationActivePowerDistributionStep.java

```diff
@@ -55,16 +56,27 @@ public List<ParticipatingElement> getParticipatingElements(Collection<LfBus> bus
                 .flatMap(bus -> bus.getGenerators().stream())
                 .filter(generator -> isParticipating(generator) && getParticipationFactor(generator) != 0)
                 .map(generator -> new ParticipatingElement(generator, getParticipationFactor(generator)))
-                .collect(Collectors.toList());
+                .collect(Collectors.toCollection(LinkedList::new));
     }
 
     @Override
-    public ActivePowerDistribution.StepResult run(List<ParticipatingElement> participatingElements, int iteration, double remainingMismatch) {
+    public double run(List<ParticipatingElement> participatingElements, int iteration, double remainingMismatch) {
         // normalize participation factors at each iteration start as some
         // generators might have reach a limit and have been discarded
         ParticipatingElement.normalizeParticipationFactors(participatingElements);
 
         double done = 0d;
+        double mismatch = remainingMismatch;
+        if (iteration == 0) {
+            // "undo" everything from targetP to go back to initialP
+            for (ParticipatingElement participatingGenerator : participatingElements) {
+                LfGenerator generator = (LfGenerator) participatingGenerator.getElement();
+                done += generator.getInitialTargetP() - generator.getTargetP();
+                mismatch -= generator.getInitialTargetP() - generator.getTargetP();
+                generator.setTargetP(generator.getInitialTargetP());
+            }
+        }
```

Comment:
ℹ️ Note to reviewers

This is the way I found to fix the issue AND also avoiding a massive refactor: today `ActivePowerDistribution` is completely _stateless_. Making `ActivePowerDistribution` aware of a "starting point" would require more work / refactorings... 

Note also that such refactoring's will need to be addressed for future developments planned in #979 and [powsybl-entsoe/#](https://github.com/powsybl/powsybl-entsoe/issues/137).
Indeed, imagine a merit order rebalancing, the starting point is crucial, otherwise you would end up with both generators from upward list moved and generators from downward list moved ...